src/main/java/uk/gov/hmcts/reform/em/stitching/rest/DocumentTaskResource.java

```diff
@@ -58,21 +62,49 @@ public DocumentTaskResource(DocumentTaskService documentTaskService) {
     public ResponseEntity<DocumentTaskDTO> createDocumentTask(
             @Valid @RequestBody DocumentTaskDTO documentTaskDTO,
             @RequestHeader(value = "Authorization", required = false) String authorisationHeader,
-            HttpServletRequest request) throws URISyntaxException {
+            HttpServletRequest request) throws URISyntaxException, IOException {
 
         log.info("REST request to save DocumentTask : {}, with headers {}", documentTaskDTO.toString(),
                 Arrays.toString(Collections.toArray(request.getHeaderNames(), new String[]{})));
 
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.registerModule(new JavaTimeModule());
+
+        String docTask = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(documentTaskDTO);
+        mapper.readValue(docTask, DocumentTaskDTO.class);
+        log.info("Received request body as Json \n" + docTask);
+
         if (documentTaskDTO.getId() != null) {
             throw new BadRequestAlertException("A new documentTask cannot already have an ID", ENTITY_NAME, "id exists");
         }
-        documentTaskDTO.setJwt(authorisationHeader);
-        documentTaskDTO.setTaskState(TaskState.NEW);
-        DocumentTaskDTO result = documentTaskService.save(documentTaskDTO);
+        DocumentTaskDTO result = null;
```

Comment:
can you initialise this in the try and return it there as well?

Reply:
we have a happy path scenario which returns result after populating it 

when moving this initialization bit inside try we get error for below block.
 return ResponseEntity.created(new URI("/api/document-tasks/" + result.getId()))
                .headers(HeaderUtil.createEntityCreationAlert(ENTITY_NAME, result.getId().toString()))
                .body(result);

Reply:
initialise it inside the try then throw the required HTTP response code with error in the catch then?